core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -442,18 +477,116 @@ private SearchResult parseToSearchResult(@Nonnull final IMAPResponse[] ir) throw
             }
             final List<Long> v = new ArrayList<Long>(); // will always return a non-null array
 
+            Long modSeq = null;
+
             // Grab all SEARCH responses
             long num;
             for (final IMAPResponse sr : ir) {
                 // There *will* be one SEARCH response.
                 if (sr.keyEquals("SEARCH")) {
                     while ((num = sr.readLong()) != -1) {
-                        v.add(Long.valueOf(num));
+                        v.add(num);
+                    }
+                    if (sr.readByte() == '(') {
```

Comment:
can it have space around

Reply:
Based on ABNF, there should be only one space before left bracket which is handled by while loop. I can still skip unnecessary spaces.

Reply:
readLong will skip spaces